### PR TITLE
Added validation for user's dob and hints to the form fields.

### LINKED ISSUE #352: Don't allow (birthdate > Date.today)
Roman suggested this. I see no issue with it.

### PR DESCRIPTION
**Pull request by pedrosino**
Fixes #352

### CODE REVIEW

File: WcaOnRails/app/models/user.rb [hunk 1]
```diff
@@ -68,6 +68,13 @@ def name_must_match_person_name
     end
   end
 
+  validate :dob_must_be_on_the_past
+  def dob_must_be_on_the_past
+    if dob.year >= Date.today.year
```
Comment (timhabermaas) [inline WcaOnRails/app/models/user.rb]: `dob >= Date.today`? Today I could enter `2016-12-31` and I'd be about -340 days old.

File: WcaOnRails/app/models/user.rb [hunk 2]
```diff
@@ -68,6 +68,13 @@ def name_must_match_person_name
...
+      errors.add(:dob, "year of birth must be on the past")
```
Comment (timhabermaas) [inline WcaOnRails/app/models/user.rb]: I think it's "in the past", but I'm not a native speaker, so take this with a grain of salt.

File: WcaOnRails/app/models/user.rb (hunk 1 above)
Comment (pedrosino) [inline WcaOnRails/app/models/user.rb]: Fixed.

File: WcaOnRails/app/models/user.rb (hunk 2 above)
Comment (pedrosino) [inline WcaOnRails/app/models/user.rb]: Fixed it, I think you're right.

**Comment by pedrosino** [comment]
OMG, Travis is complaining a lot...

**Comment by pedrosino** [comment]
Of course Travis was complaining...it was trying to compare nil to Date.today :P
I fixed it.

File: WcaOnRails/app/views/users/edit.html.erb [hunk 3]
```diff
@@ -75,7 +75,7 @@
     <% end %>
 
     <%= f.input :name, disabled: !editable_fields.include?(:name) %>
-    <%= f.input :dob, as: :date_picker, disabled: !editable_fields.include?(:dob) %>
+    <%= f.input :dob, as: :date_picker, disabled: !editable_fields.include?(:dob), hint: "Enter your date of birth in the format YYYY-MM-DD" %>
```
Comment (jfly) [inline WcaOnRails/app/views/users/edit.html.erb]: I'm not in love with copying this around, would it be possible to do this instead by changing en.yml as per http://stackoverflow.com/a/23649169/1739415? Maybe something like this:

```yml
en:
  simple_form:
    hints:
      defaults:
        dob: 'Enter your date of birth in the format YYYY-MM-DD'
```

**Comment by jfly** [comment]
Thanks for taking this on, Pedro! The validation looks good, just one question about code duplication for you.

File: WcaOnRails/app/views/users/edit.html.erb (hunk 3 above)
Comment (pedrosino) [inline WcaOnRails/app/views/users/edit.html.erb]: Ok, fixed.

**Comment by jfly** [comment]
LGTM, merged!

**Pull request merged**